apps/wallet/src/ui/app/pages/swap/index.tsx

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { useRecognizedPackages } from '_app/hooks/useRecognizedPackages';
+import { useSigner } from '_app/hooks/useSigner';
+import BottomMenuLayout, { Content, Menu } from '_app/shared/bottom-menu-layout';
+import { Button } from '_app/shared/ButtonUI';
+import { Form } from '_app/shared/forms/Form';
+import { InputWithActionButton } from '_app/shared/InputWithAction';
+import Loading from '_components/loading';
+import Overlay from '_components/overlay';
+import { filterAndSortTokenBalances } from '_helpers';
+import {
+	allowedSwapCoinsList,
+	Coins,
+	getUSDCurrency,
+	isExceedingSlippageTolerance,
+	SUI_CONVERSION_RATE,
+	USDC_DECIMALS,
+	useBalanceConversion,
+	useCoinsReFetchingConfig,
+	useGetEstimate,
+	useMainnetCoinsMap,
+	useMainnetPools,
+	useSortedCoinsByCategories,
+} from '_hooks';
+import {
+	DeepBookContextProvider,
+	useDeepBookAccountCapId,
+	useDeepBookClient,
+} from '_shared/deepBook/context';
+import { useFormatCoin, useTransactionSummary, useZodForm } from '@mysten/core';
+import { useSuiClientQuery } from '@mysten/dapp-kit';
+import { ArrowDown12, ArrowRight16 } from '@mysten/icons';
+import { type DryRunTransactionBlockResponse } from '@mysten/sui.js/client';
+import { SUI_DECIMALS, SUI_TYPE_ARG } from '@mysten/sui.js/utils';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import BigNumber from 'bignumber.js';
+import clsx from 'classnames';
+import { useMemo } from 'react';
+import { useWatch, type SubmitHandler } from 'react-hook-form';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { z } from 'zod';
+
+import { AssetData } from './AssetData';
+import { GasFeeSection } from './GasFeeSection';
+import { ToAssetSection } from './ToAssetSection';
+import { initialValues, type FormValues } from './utils';
+
+function getSwapPageAtcText(
+	fromSymbol: string,
+	toAssetType: string,
+	coinsMap: Record<string, string>,
+) {
+	const toSymbol =
+		toAssetType === SUI_TYPE_ARG
+			? Coins.SUI
+			: Object.entries(coinsMap).find(([key, value]) => value === toAssetType)?.[0] || '';
+
+	return `Swap ${fromSymbol} to ${toSymbol}`;
+}
+
+export function SwapPageContent() {
+	const queryClient = useQueryClient();
+	const mainnetPools = useMainnetPools();
+	const navigate = useNavigate();
+	const [searchParams] = useSearchParams();
+	const activeAccount = useActiveAccount();
+	const signer = useSigner(activeAccount);
+	const activeAccountAddress = activeAccount?.address;
+	const { staleTime, refetchInterval } = useCoinsReFetchingConfig();
+	const coinsMap = useMainnetCoinsMap();
+	const deepBookClient = useDeepBookClient();
+
+	const accountCapId = useDeepBookAccountCapId();
```

Comment:
This is actually interestingly slightly wrong and error prone, because the account cap could have been failed to fetch within the provider. I honestly would probably go back to dynamic deepbook SDK construction when you construct orders, to avoid that issue.